c10/core/TensorImpl.cpp

```diff
@@ -83,6 +83,7 @@ bool TensorImpl::compute_contiguous() const {
 }
 
 void TensorImpl::release_resources() {
+  autograd_meta_.reset();
```

Comment:
The old code is "sound" because `Variable::Impl` destructor was responsible for clearing it. I remember discussing this with Will but I don't remember how he convinced me that we shouldn't "fix" it for now. In any case, all good now.